### PR TITLE
new_detections only as comments, determine the label of the tracks wi…

### PR DESCRIPTION
**Pull request by arminkollascheck**
…th a dataframe (.mode) instead of .count function (way faster), dont copy informations of the last assigned detection of the track during gaps

Only returns the "tracks_finished" instead of "new_detections, tracks_finished". Could be a problem for functions that use the track_iou() function, even if they dont use the new_detections.

### CODE REVIEW

**Comment by martinbaerwolff** [comment]
@frunika do u want to discard the outcommented sections?

**Comment by martinbaerwolff** [comment]
@arminkollascheck thanks for tremendously boosting the pipeline ;)

File: OTVision/track/iou_tracker_qp.py [hunk 1]
```diff
@@ -125,13 +125,13 @@ def track_iou(detections, sigma_l, sigma_h, sigma_iou, t_min, t_miss_max):
...
+    # for frame_num in new_detections:
+    #     for det in new_detections[frame_num]["classified"]:
+    #         if det["vehID"] not in vehIDs_finished:
+    #             det["finished"] = False
+    #         else:
+    #             det["finished"] = True
+    #             det['label'] = tracks[tracks['vehID'] == det['vehID']]['max_label']
 
     return tracks_finished
     # return new_detections, tracks_finished
```
Comment (arminkollascheck) [inline OTVision/track/iou_tracker_qp.py]: also if you want the change it to the old with new_detections

Comment (arminkollascheck) [reply OTVision/track/iou_tracker_qp.py]: and line 44, 50 and the 74 following

**Review by arminkollascheck — COMMENTED** [review]
If you want to reverse the new_detections change.

File: OTVision/track/iou_tracker_qp.py [hunk 2]
```diff
@@ -71,23 +71,19 @@ def track_iou(detections, sigma_l, sigma_h, sigma_iou, t_min, t_miss_max):
 
                     # remove best matching detection from detections
                     del dets[dets.index(best_match)]
-                    best_match["vehID"] = track["vehID"]
+                    """best_match["vehID"] = track["vehID"]
```
Comment (frunika) [inline OTVision/track/iou_tracker_qp.py:74]: Don't use docstrings as comment. Use line comment (#)

File: OTVision/track/iou_tracker_qp.py [hunk 3]
```diff
@@ -110,31 +106,35 @@ def track_iou(detections, sigma_l, sigma_h, sigma_iou, t_min, t_miss_max):
...
-        if track["max_confidence"] >= sigma_h and len(track["bboxes"]) >= t_min
+        if (
+            track["max_confidence"] >= sigma_h 
+            and track["frames"][-1] - track["frames"][0] >= t_min
+        )
     ]
 
     for track in tracks_finished:
-        track["max_label"] = max(track["labels"], key=track["labels"].count)
+        track["max_label"] = pd.Series(track["labels"]).mode().iat[0]
```
Comment (frunika) [inline OTVision/track/iou_tracker_qp.py:125]: nice one ;)

**Review by frunika — CHANGES_REQUESTED** [review]
just some minor changes due to docstring usage :)

Thanks for implementation @arminkollascheck

**Comment by arminkollascheck** [comment]
Ok

**Pull request merged**